src/Aspire.Hosting.Azure.AppContainers/AzureContainerAppsInfrastructure.cs

```diff
@@ -38,7 +38,12 @@ public async Task BeforeStartAsync(DistributedApplicationModel appModel, Cancell
             throw new NotSupportedException("Multiple container app environments are not supported.");
         }
 
-        var environment = caes.FirstOrDefault() as IAzureContainerAppEnvironment ?? new AzdAzureContainerAppEnvironment();
+        var environment = caes.FirstOrDefault();
+
+        if (environment is null)
+        {
+            return;
```

Comment:
Should this throw?

Reply:
Also you can remove the comment on line 73:

```
ComputeEnvironment = environment as IComputeEnvironmentResource // will be null if azd
```

And maybe the cast too?

And fix line 72, it should be `ContainerRegistry = environment,`.